contracts/src/periphery/EigenDACertVerifierV1.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifierV1} from "src/periphery/interfaces/IEigenDACertVerifierV1.sol";
+import {IEigenDAThresholdRegistry} from "src/interfaces/IEigenDAThresholdRegistry.sol";
+import {IEigenDABatchMetadataStorage} from "src/interfaces/IEigenDABatchMetadataStorage.sol";
+import {EigenDACertVerificationV1Lib as CertLib} from "src/periphery/libraries/EigenDACertVerificationV1Lib.sol";
+import "src/interfaces/IEigenDAStructs.sol";
+
+/**
+ * @title CertVerifierV1 - EigenDA V1 certificate verification
+ * @author Layr Labs, Inc.
+ * @notice Contains all V1-specific verification functionality
+ */
+contract EigenDACertVerifierV1 is IEigenDACertVerifierV1 {
+    /// @notice Thrown when there is a length mismatch
+    error LengthMismatch();
+
+    /// @notice The EigenDAThresholdRegistry contract address
+    IEigenDAThresholdRegistry public immutable eigenDAThresholdRegistryV1;
+
+    /// @notice The EigenDABatchMetadataStorage contract address
+    /// @dev On L1 this contract is the EigenDA Service Manager contract
+    IEigenDABatchMetadataStorage public immutable eigenDABatchMetadataStorageV1;
+
+    /**
+     * @notice Constructor for the EigenDA V1 certificate verifier
+     * @param _eigenDAThresholdRegistryV1 The address of the EigenDAThresholdRegistry contract
+     * @param _eigenDABatchMetadataStorageV1 The address of the EigenDABatchMetadataStorage contract
+     */
+    constructor(
+        IEigenDAThresholdRegistry _eigenDAThresholdRegistryV1,
+        IEigenDABatchMetadataStorage _eigenDABatchMetadataStorageV1
+    ) {
+        eigenDAThresholdRegistryV1 = _eigenDAThresholdRegistryV1;
+        eigenDABatchMetadataStorageV1 = _eigenDABatchMetadataStorageV1;
+    }
+
+    /**
+     * @notice Verifies a the blob cert is valid for the required quorums
+     * @param blobHeader Pointer to the blob header in calldata
+     * @param blobVerificationProof Pointer to the blob cert verification proof in calldata
+     */
+    function verifyDACertV1(BlobHeader calldata blobHeader, BlobVerificationProof calldata blobVerificationProof)
+        public
+        view
+    {
+        (CertLib.ErrorCode err, bytes memory errParams) = CertLib.verifyDACert(
+            _quorumConfirmationThresholdPercentages(),
+            _storedBatchMetadataHash(blobVerificationProof),
+            blobHeader,
+            blobVerificationProof,
+            _quorumNumbersRequired()
+        );
+        CertLib.revertOnError(err, errParams);
+    }
+
+    /**
+     * @notice Verifies a batch of blob certs for the required quorums
+     * @param blobHeaders Pointer to array of blob headers in calldata
+     * @param blobVerificationProofs Pointer to array of blob cert verification proofs in calldata
+     */
+    function verifyDACertsV1(BlobHeader[] calldata blobHeaders, BlobVerificationProof[] calldata blobVerificationProofs)
```

Comment:
> I don't think anyone is using this afaiu. Should we just remove? This also feels like something that can be provided by some outer layer like a multicall contract right?

I agree. However, I think removals can be done in another PR as another breaking change, to limit the already pretty large scope of this PR.

Reply:
I agree. However, I think removals can be done in another PR as another breaking change, to limit the already pretty large scope of this PR.

Reply:
Sure, I can remove this, and this would also remove length mismatch too.

Reply:
https://github.com/Layr-Labs/eigenda/pull/1477/commits/c7963d237e43d0cb722f1bad8bbe4fef35d933a4